File: Creational/FactoryMethod/FactoryMethod/BikeFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FactoryMethod
{
    class BikeFactory : Transport
    {
        /// <summary>
        /// Concrete Creators override the base factory method so it returns
        /// a different type of product.
        /// </summary>
        /// <returns>A new type of product</returns>
        protected override IVehicle CreateTransport()
        {
            return new Bike();
        }

        /* Note that the factory method doesn’t have to 
         * create new instances all the time. It can also
         * return existing objects from a cache, an object pool, or another source.
         */
    }
}

File: Creational/FactoryMethod/TypeScript/src/factories/bike-factory.ts
import { Bike } from '../vehicle/bike';
import { Vehicle } from '../vehicle/vehicle';
import { VehicleFactory } from './vehicle-factory';

// Concrete creator
export class BikeFactory extends VehicleFactory {
  getVehicle(vehicleName: string): Vehicle {
    return new Bike(vehicleName);
  }
}
